mapper.cpp

```diff
@@ -474,7 +532,31 @@ vector<Alignment> Mapper::align_threaded(Alignment& alignment, int& kmer_count,
         // Grab the map from node ID to kmer start positions for this particular kmer.
         auto& kmer_positions = positions.at(i);
         // Fill it in, since we know there won't be too many to work with.
-        index->get_kmer_positions(k, kmer_positions);
+        
+        if(gcsa) {
+            // We need to fill in this vector with the GCSA nodes and then convert.
+            std::vector<gcsa::node_type> gcsa_nodes;
+            gcsa->locate(gcsa_range, gcsa_nodes);
+            
+            for(gcsa::node_type gcsa_node : gcsa_nodes) {
+                if(gcsa::Node::rc(gcsa_node)) {
+                    // We found a kmer on the reverse strand. The old index
```

Comment:
Yeah, this is a little confusing. There may be situations where we use a forward-only index. (I'd like there not to be but it's not clear yet.)